src/run_tests.sh

```diff
@@ -42,11 +42,22 @@ VTESTS=`find validation_tests/* -prune -perm -u+x -type f ! -name "*.[c|h]"`;
 #CTESTS=`find ctests -maxdepth 1 -perm -u+x -type f`;
 CTESTS=`find ctests/* -prune -perm -u+x -type f ! -name "*.[c|h]"`;
 FTESTS=`find ftests -perm -u+x -type f ! -name "*.[c|h|F]"`;
-COMPTESTS=`find components/*/tests -perm -u+x -type f ! \( -name "*.[c|h]" -o -name "*.cu" -o -name "*.so" \)`;
-#EXCLUDE=`grep --regexp=^# --invert-match run_tests_exclude.txt`
-EXCLUDE=`grep -v -e '^#\|^$' run_tests_exclude.txt`
 
 
+
+# List of active components
+ACTIVE_COMPONENTS_PATTERN=$(papi_component_avail | awk '/Active components:/{flag=1; next} flag' | grep "Name:" | sed 's/Name: //' | awk '{print $1}' | paste -sd'|' -)
```

Comment:
We should prefix 'papi_component_avail' with 'utils/' since it may not be installed yet. What do you think @jagode ?

Reply:
Great. We will go ahead and rebase and merge.

Reply:
Upon further testing, I get the following error when I configure-in the CUDA component:
>Running Component Tests
>Running components/cuda/tests/test_2thr_1gpu_not_allowed:
>    ./components/cuda/tests/test_2thr_1gpu_not_allowed: error while loading shared
>    libraries: libpfm.so.4: cannot open shared object file: No such file or directory

Since the shared object file libpfm.so.4 is contained in src/libpfm4/lib, one solution would be to prepend the environment variables in run_test.sh
>LD_LIBRARY_PATH=.:./libpfm-3.y/lib:"$LD_LIBRARY_PATH"
>LIBPATH=.:./libpfm-3.y/lib:"$LIBPATH"

with ./libpfm4/lib.

Perhaps this warrants a separate pull request, though, as it appears to be a separate issue @jagode?

Reply:
It does seem to be a seperate issue after the libpfm4 changes were merged into the master branch. I could look into the issue and create a seperate PR soon after. @dbarry9 @jagode.